### PR TITLE
Document Default controller in routing docs

### PR DESCRIPTION
**Pull request by mikem**
I found the `Default` controller useful and thought it'd be good to mention it in the routing documentation. This PR does that, along with some other minor documentation fixes.

### CODE REVIEW

**Comment by mikem** [comment]
Oh, looks like the docs are compiled, too. I took a look, but I'm running out of time to dig into how to set this straight. Any pointers welcome :-)

**Comment by wsargent** [comment]
You need to specify the full path to the controller type in routes:

```
[error] /home/travis/build/playframework/playframework/documentation/manual/working/javaGuide/main/http/code/javaguide.http.routing.defaultcontroller.routes:3: type Default is not a member of package javaguide.http.routing.defaultcontroller.controllers
[error] GET   /about      controllers.Default.redirect(to = "https://www.playframework.com/")
[error] 
```

https://travis-ci.org/playframework/playframework/builds/157275557#L3208

**Comment by mikem** [comment]
@wsargent hm, but I'd rather the code samples in the documentation show exactly how one would use the `Default` controller in their own project.

I managed to get the build passing in c47c40622b5e83760bd824f140c89b5cd68c520b. There are a couple of things I'd like to improve, but can't figure out how. See my comments below.

File: documentation/manual/working/javaGuide/main/http/code/javaguide/http/JavaRouting.scala [hunk 1]
```diff
@@ -106,3 +123,16 @@ class Clients extends Controller {
 }
 }
 
+package routing.defaultcontroller.controllers {
+import play.api.mvc.Controller
+import _root_.controllers.{Default => PlayDefault}
+
+class Default extends Controller {
```
Comment (mikem) [inline documentation/manual/working/javaGuide/main/http/code/javaguide/http/JavaRouting.scala]: It'd be great not to have to declare a new controller myself here, and just import the `Default` controller and be done with it.

File: documentation/manual/working/scalaGuide/main/http/code/ScalaRouting.scala [hunk 2]
```diff
@@ -72,6 +72,20 @@ package defaultvalue.controllers {
   }
 }
 
+package defaultcontroller.controllers {
+  import _root_.controllers.{Default => PlayDefault}
+
+  class Default extends Controller {
```
Comment (mikem) [inline documentation/manual/working/scalaGuide/main/http/code/ScalaRouting.scala]: Here too, I'd like to just have the import, and not declare my own `Default` controller.

File: documentation/manual/working/scalaGuide/main/http/code/ScalaRouting.scala [hunk 3]
```diff
@@ -151,4 +171,17 @@ object ScalaRoutingSpec extends Specification {
...
+
+  def statusOf(rh: RequestHeader, router: Class[_ <: Router] = classOf[Routes]) = {
+    running() { app =>
+      implicit val mat = ActorMaterializer()(app.actorSystem)
+      status {
+        val routedHandler = app.injector.instanceOf(router).routes(rh)
+        val (rh2, terminalHandler) = Handler.applyStages(rh, routedHandler)
+        terminalHandler match {
+          case e: EssentialAction => e(rh2).run()
+        }
```
Comment (mikem) [inline documentation/manual/working/scalaGuide/main/http/code/ScalaRouting.scala]: I ran out of time trying to get the common bits of `contentOf` and `statusOf` extracted into a function.

**Comment by mikem** [comment]
The [latest build failure](https://travis-ci.org/playframework/playframework/builds/157409117) seems unrelated to the documentation.

File: documentation/manual/working/javaGuide/main/http/code/javaguide/http/JavaRouting.scala (hunk 1 above)
Comment (gmethvin) [inline documentation/manual/working/javaGuide/main/http/code/javaguide/http/JavaRouting.scala]: @mikem Why not just

```scala
class Default extends _root_.controllers.Default
```
?

It's somewhat annoying that we have to do this. I never really liked the fact that we have stuff in a top-level `controllers` package either (I would rather it be `play.controllers` or something).

**Comment by mikem** [comment]
@gmethvin thanks for the suggestion -- I applied it in 2614316e2a55af6591bacab12cada7e30b76c984.

I also rebased my branch on top of the current master.

**Pull request merged**